nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/SetHealthCheckConfiguration.java

```diff
@@ -120,7 +120,7 @@
     private String target;
 
     @Param(name = "enabled")
-    private boolean enabled;
+    private Boolean enabled;
```

Comment:
While `boolean` does work mostly as expected it has one unexpected behaviour: The `enabled` parameter effectively becomes optional as HK2 does not complain about a missing parameter and simply will use `false`.

Reply:
What is the behavior with boxed boolean?

If it is `null` when parameter is not present, then I see a `NullPointerException` in `updateConfig` and `updateLogNotifier`

Reply:
This is confusing HK2 magic. The parameter is not optional thus you only ever get so actually running the command if it is provided. But if you make it `boolean` instead of `Boolean` it behaves as if the parameter is optional with default `false`. This is why I changed this back to `Boolean` as we want the console to ask you for the `enabled` parameter if you forget to provide it.